protocol/auditlog.go

```diff
@@ -100,6 +116,7 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 	// create the new directory history
 	h = newDirectoryHistory(addr, signKey, snaps[0])
 
+	// FIXME: remove this check --> caller calls Audit() before this function
```

Comment:
So, what I'm suggesting is

```
Audit(Reponse r) {
  pass = AuditDirectory()
  if pass {
    if new directory
      insert(snaps)
    else
      update(snaps)
  }
}
```
I haven't implemented this yet, because it's part of the server-auditor protocol.